l2geth/core/state_transition.go

```diff
@@ -122,6 +125,15 @@ func IntrinsicGas(data []byte, contractCreation, isHomestead bool, isEIP2028 boo
 
 // NewStateTransition initialises and returns a new state transition object.
 func NewStateTransition(evm *vm.EVM, msg Message, gp *GasPool) *StateTransition {
+	l1Fee := new(big.Int)
+	if rcfg.UsingOVM {
+		if msg.GasPrice().Cmp(common.Big0) != 0 {
+			// Compute the L1 fee before the state transition
+			// so it only has to be read from state one time.
+			l1Fee, _ = fees.CalculateL1MsgFee(msg, evm.StateDB, nil)
```

Comment:
Consider putting base case inside of `fees.CalculateL1MsgFee` where if `msg.GasPrice() == 0` then return `0` to remove line 130

Reply:
I prototyped this and its going to require a bit of a refactor to the tests to get this working like the above comment. Willing to do so if strong preference